ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/ruleeditorhook.js

```diff
@@ -32,22 +32,93 @@
         ruleEditorFrame.setAttribute('id','af-rule-editor');
         let formContainerPath = getFormContainerPath(editable);
         if (!formContainerPath) {
-            showAlert();
+            if (Granite.Toggles.isEnabled("FT_FORMS-14068")) {
```

Comment:
Core components should not reference any toggle API; ideally, this code should reside in cq-guides, and we should simply reference it here. Customers won't customize this behavior.

Reply:
I found two instances of using toggles in core component:

- Granite: https://github.com/adobe/aem-core-forms-components/blob/6d4cd26768f9b68c8c4ade5487deefc66da099f6/ui.apps/src/main/content/jcr_root/apps/core/fd/components/aemform/v2/aemform/clientlibs/editorhook/js/EditListeners.js#L54-L56

- toggles.json api: https://github.com/adobe/aem-core-forms-components/blob/6d4cd26768f9b68c8c4ade5487deefc66da099f6/ui.apps/src/main/content/jcr_root/apps/core/fd/components/aemform/v1/aemform/clientlibs/runtime/formapp/js/formapp.js#L35-L40

can we api call approach or go with cq-guides only?

Reply:
this is old code which was moved long back from cq-guides to core components, I would prefer this code to move to cq-guides. It would be dependent on FAR changes if you move to cq-guides